src/C++/SSLSocketAcceptor.cpp

```diff
@@ -365,6 +371,12 @@ bool SSLSocketAcceptor::onData( SocketServer& server, socket_handle s )
   SocketConnections::iterator i = m_connections.find( s );
   if ( i == m_connections.end() ) return false;
   SSLSocketConnection* pSocketConnection = i->second;
+
+  if (pSocketConnection->didProcessQueueRequestToRead()) {
+      pSocketConnection->processQueue();
+      pSocketConnection->signal();
+  }
+
```

Comment:
When processQueue() is called, it may be that SSL_write fails because the OpenSSL statemechine wants to read in order to renegotiate keys. In that case we raise this other flag to indicate that processQueue requested to read.
In this method (onData) we check if that flag is set, we call processQueue again to resume the renegotiation, and continue the write.